Src/Particle/AMReX_Particle.H

```diff
@@ -102,6 +102,28 @@ struct ParticleIDWrapper
         r = (sign) ? lval : -lval;
         return r;
     }
+
+    /** Swap the sign of the id.
+     *
+     * Positive ids become negative and negative ids become positive.
+     */
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    void negate () noexcept
+    {
+        // the leftmost bit is our id sign - we will XOR it with 1 to flip it
+        m_idata ^= static_cast<uint64_t>(1) << 63;
+    }
+
+    /** Check the sign of the id.
+     *
+     * Returns false if the id is positive, otherwise true (invalid particle).
+     */
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    bool negative () noexcept
+    {
+        // the leftmost bit is our id sign
+        return value >> 63;
```

Comment:
According to godbolt, I think I wrote the write test here (if the bit is 1 it is negative).

I am confused with the `Long ()` implementation above reading:
```C++
        Long lval = static_cast<Long>(val);  // bc we take -
        r = (sign) ? lval : -lval;
```

Shouldn´t that read
```C++
        r = (sign) ? lval : -lval;
```
?